### PR TITLE
Autobuild website on Travis from stable branch

### PR DESCRIPTION
**Pull request by spicyj**
$REACT_WEBSITE_BRANCH in https://travis-ci.org/facebook/react/settings/env_vars now needs to point to the stable branch (currently 0.13-stable). I haven't tested the commit-and-push part of this but everything else works so I'm hopeful.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,16 +5,51 @@ node_js:
...
+        GH_PAGES_DIR="$TRAVIS_BUILD_DIR"/../react-gh-pages
+        echo "machine github.com login reactjs-bot password $GITHUB_TOKEN" >~/.netrc
+
+        # TODO: Can't actually use the Travis cache because shrinkwrapped deps
+        # won't override those already installed. (Without this, the built
+        # JSXTransformer.js uses different dependencies and makes a different
+        # built file.)
+        rm -rf node_modules
+        npm install
+        grunt build  # to create docs/js/{react,JSXTransformer}.js
```
Comment (zpao) [inline .travis.yml]: I was wondering about this step. Would it maybe make sense to just exclude these always? Then we intentionally check the right ones in when we ship a new release?

File: .travis.yml [hunk 2]
```diff
@@ -39,12 +74,15 @@ env:
...
   - TEST_TYPE=lint
+  - TEST_TYPE=build_website
   - TEST_TYPE=test:webdriver:saucelabs:modern
   global:
   # SERVER
   - secure: qPvsJ46XzGrdIuPA70b55xQNGF8jcK7N1LN5CCQYYocXLa+fBrl+fTE77QvehOPhqwJXcj6kOxI+sY0KrVwV7gmq2XY2HZGWUSCxTN0SZlNIzqPA80Y7G/yOjA4PUt8LKgP+8tptyhTAY56qf+hgW8BoLiKOdztYF2p+3zXOLuA=
   # SECRET_TOKEN
   - secure: dkpPW+VnoqC/okhRdV90m36NcyBFhcwEKL3bNFExAwi0dXnFao8RoFlvnwiPlA23h2faROkMIetXlti6Aju08BgUFV+f9aL6vLyU7gUent4Nd3413zf2fwDtXIWIETg6uLnOpSykGKgCAT/hY3Q2oPLqOoY0OxfgnbqwxkxljrE=
+  # GITHUB_TOKEN
+  - secure: EHCyCSKMwKlLHNtcj9nmkRzmiiPE3aDGlPcnEyrDJeRI0SeN/iCXHXfFivR0vFq3vr+9naMBczAR2AEidtps5KbJrKqdZnjPFRbmfVtzWr/LlvVCub3u13Pub6TdKIVBTny1PuZ5X8GvdxMNVig89jGjvzhhWuQRaz3VhJnTra4=
```
Comment (zpao) [inline .travis.yml]: I don't know the difference between storing them here vs on Travis itself. I put the first few in here but this PR is mixing storage for new ones. Any reason?

Comment (spicyj) [inline .travis.yml]: I figured we'd put ones here by default but REACT_WEBSITE_BRANCH we want to be able to change and have it affect all branches (so if we push new commits to an old -stable branch) so I did it in the web UI.

File: .travis.yml (hunk 1 above)
Comment (spicyj) [inline .travis.yml]: Yeah, that might be better.

**Pull request merged**